src/main/java/org/sagebionetworks/bridge/sdk/ClientProvider.java

```diff
@@ -36,6 +36,9 @@ public static ClientInfo getClientInfo() {
      */
     public static Session signIn(SignInCredentials signIn) {
         checkNotNull(signIn, "SignInCredentials required.");
+        checkArgument(isNotBlank(signIn.getStudyIdentifier()), "Study identifier cannot be blank/null");
+        checkArgument(isNotBlank(signIn.getUsername()), "Username cannot be blank/null");
+        checkArgument(isNotBlank(signIn.getPassword()), "Password cannot be blank/null");
```

Comment:
Would it be better to move the null checks to the SignInCredentials constructor? This way, you can guarantee that SignInCredentials will always have non-blank fields. Similarly for SignUp and Email.